flake.nix

```diff
@@ -27,206 +28,267 @@ rec {
   };
 
   outputs =
-    { self
-    , nixpkgs
-    , flake-utils
-    , released-nixpkgs-unstable
-    , released-nixpkgs-stable
-    , released-nix-unstable
-    , released-nix-stable
-    , nix-pills
+    inputs@{
+      self,
+      flake-parts,
+      git-hooks-nix,
+      nixpkgs,
+      released-nixpkgs-unstable,
+      released-nixpkgs-stable,
+      released-nix-unstable,
+      released-nix-stable,
+      nix-pills,
+      systems,
     }:
-    flake-utils.lib.eachDefaultSystem (system:
-      let
-        overlay = final: prev: {
-          asciinema-scenario = final.rustPlatform.buildRustPackage rec {
-            pname = "asciinema-scenario";
-            version = "0.2.0";
-            src = final.fetchCrate {
-              inherit pname version;
-              sha256 = "sha256-qMGi+myppWBapM7TkPeXC2g/M1FA1YGwESNrx8LVXkw=";
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      imports = [
+        inputs.git-hooks-nix.flakeModule
+      ];
+      systems = import systems;
```

Comment:
You forgot to import systems in the inputs attribute (https://github.com/loophp/nix-shell/blob/4a8220125e9537dfa316b49e2b9cf77e977c86af/flake.nix#L9).